### PR TITLE
Fixes #4746 inc/functions_search.php warnings

### LINKED ISSUE #4746: inc/functions_search.php warnings
https://community.mybb.com/thread-239129.html

### PR DESCRIPTION
**Pull request by Eldenroot**
Resolves #4746

### CODE REVIEW

**Comment by HLFadmin** [comment]
Typo in line 1564. Should be
if(!empty($search['postdate']))

File: inc/functions_search.php [hunk 1]
```diff
@@ -1560,7 +1561,7 @@ function perform_search_mysql_ft($search)
 		}
 	}
 	$datecut = $thread_datecut = $post_datecut = '';
-	if($search['postdate'])
+	if(!empty($search['author']))
```
Comment (Stefan-MyBB) [inline inc/functions_search.php]: This is obviously incorrect. ;)

File: inc/functions_search.php [hunk 2]
```diff
@@ -1578,7 +1579,7 @@ function perform_search_mysql_ft($search)
 	}
 
 	$thread_replycut = '';
-	if($search['numreplies'] != '' && $search['findthreadst'])
+	if(isset($search['numreplies']) && $search['numreplies'] != '' && $search['findthreadst'])
```
Comment (Stefan-MyBB) [inline inc/functions_search.php]: Using !empty here would be shorter.

File: inc/functions_search.php [hunk 3]
```diff
@@ -1592,7 +1593,7 @@ function perform_search_mysql_ft($search)
 
 	$thread_prefixcut = '';
 	$prefixlist = array();
-	if($search['threadprefix'] && $search['threadprefix'][0] != 'any')
+	if(!empty($search['threadprefix']) && $search['threadprefix'][0] != 'any')
```
Comment (Stefan-MyBB) [inline inc/functions_search.php:1596]: It might be sensible to check that $search['threadprefix'] is an array unless this is already done somewhere else.

**Review by Stefan-MyBB — CHANGES_REQUESTED** [review]

File: inc/functions_search.php (hunk 1 above)
Comment (Eldenroot) [reply inc/functions_search.php]: Fixed.

File: inc/functions_search.php (hunk 2 above)
Comment (Sama34) [inline inc/functions_search.php]:
```suggestion
	if(!empty($search['numreplies']) && $search['findthreadst'])
```

File: inc/functions_search.php (hunk 3 above)
Comment (Sama34) [reply inc/functions_search.php:1596]: When set, it is assumed to be an array  :
https://github.com/mybb/mybb/blob/f18576b25de6c4f67501fe54910f15f2b7769bd2/search.php#L1587

If anything the following should be updated for consistency :
https://github.com/mybb/mybb/blob/f18576b25de6c4f67501fe54910f15f2b7769bd2/inc/functions_search.php#L1241

**Review by Sama34 — CHANGES_REQUESTED** [review]

File: inc/functions_search.php (hunk 3 above)
Comment (Eldenroot) [reply inc/functions_search.php:1596]: This is fixed, thanks!

File: inc/functions_search.php (hunk 2 above)
Comment (Eldenroot) [reply inc/functions_search.php]: Fixed. Thanks!

File: inc/functions_search.php (hunk 3 above)
Comment (Eldenroot) [reply inc/functions_search.php:1596]: Fixed. Thanks!

**Pull request merged**